go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -321,110 +326,124 @@ func (m *MasterController) setupHybridLRPolicySharedGw(nodeSubnets []*net.IPNet,
 		} else {
 			L3Prefix = "ip4"
 		}
-		var hybridCIDR *net.IPNet
-		for _, hybridSubnet := range config.HybridOverlay.ClusterSubnets {
-			if utilnet.IsIPv6CIDR(hybridSubnet.CIDR) == utilnet.IsIPv6CIDR(nodeSubnet) {
-				hybridCIDR = hybridSubnet.CIDR
-				break
+		var hybridCIDRs []*net.IPNet
+		if len(config.HybridOverlay.ClusterSubnets) > 0 {
+			for _, hybridSubnet := range config.HybridOverlay.ClusterSubnets {
+				if utilnet.IsIPv6CIDR(hybridSubnet.CIDR) == utilnet.IsIPv6CIDR(nodeSubnet) {
+					hybridCIDRs = append(hybridCIDRs, hybridSubnet.CIDR)
+					break
+				}
+			}
+		} else {
+			nodes, err := m.kube.GetNodes()
+			if err != nil {
+				return err
+			}
+			for _, node := range nodes.Items {
+				if subnet, _ := houtil.ParseHybridOverlayHostSubnet(&node); subnet != nil {
+					hybridCIDRs = append(hybridCIDRs, subnet)
+				}
 			}
 		}
 
-		drIP := util.GetNodeHybridOverlayIfAddr(nodeSubnet).IP
-		matchStr := fmt.Sprintf(`inport == "%s%s" && %s.dst == %s`,
-			ovntypes.RouterToSwitchPrefix, nodeName, L3Prefix, hybridCIDR)
-
-		// Logic route policy to steer packet from pod to hybrid overlay nodes
-		logicalRouterPolicy := nbdb.LogicalRouterPolicy{
-			Priority: ovntypes.HybridOverlaySubnetPriority,
-			ExternalIDs: map[string]string{
-				"name": ovntypes.HybridSubnetPrefix + nodeName,
-			},
-			Action:   nbdb.LogicalRouterPolicyActionReroute,
-			Nexthops: []string{drIP.String()},
-			Match:    matchStr,
-		}
+		for _, hybridCIDR := range hybridCIDRs {
+			drIP := util.GetNodeHybridOverlayIfAddr(nodeSubnet).IP
```

Comment:
dont you need to check if hte hybridCIDR matches the same IP family as the node subnet like the code that iterates through the HO cluster subnets does, i.e.:

if utilnet.IsIPv6CIDR(hybridSubnet.CIDR) == utilnet.IsIPv6CIDR(nodeSubnet)

Reply:
I don't think we support IPv6 with hybrid overlay, do we?

Reply:
as of now we do not, it is something that we are looking to add

Reply:
fixed.